main.tf

```
@@ -287,9 +287,13 @@ resource "aws_ecr_registry_scanning_configuration" "this" {
     content {
       scan_frequency = rule.value.scan_frequency
 
-      repository_filter {
-        filter      = rule.value.filter
-        filter_type = try(rule.value.filter_type, "WILDCARD")
+      dynamic "repository_filter" {
+        for_each = rule.value.filters
```

I know the plural form makes more sense, but it does throw users off - lets keep the current, singular form to avoid surprising folks:
```suggestion
        for_each = rule.value.filter
```

---

Added suggested change